src/phone_formatter.js

```diff
@@ -197,9 +261,32 @@ class PhoneFormatter extends DelimitedTextFormatter {
    * @private
    */
   guessFormatFromText(text) {
+    currentCountryCode = DEFAULT_COUNTRY_CODE;
     if (text && text[0] === '+') {
-      this.delimiterMap = NANPPhoneDelimitersWithPlus;
-      this.maximumLength = 1 + 1 + 10 + 5;
+      if (text.length > 1) {
+        const isValidCountryCode = countryCode => COUNTRY_CODES.reduce(
+          (memo, obj) => memo.E164 === countryCode && memo ||
+            obj.E164 === countryCode && obj
```

Comment:
This `reduce` function feels a little too clever for me. Also, it implies that it's going to return a `boolean` but it actually returns an object. If all it's doing is mapping a string to a country and some options, why not make `COUNTRY_CODES` a map?